coremltools/converters/mil/mil/passes/tests/test_passes.py

```diff
@@ -1674,6 +1674,41 @@ def prog(x):
             backend=backend,
         )
 
+    @pytest.mark.parametrize(
+        "backend",
+        backends,
+    )
+    def test_merge_reshape_in_nested_block(self, backend):
+        INPUT_SHAPE = (6, 7)
+        OUTPUT_SHAPE = (14, 3)
+
+        @mb.program(input_specs=[mb.TensorSpec(shape=INPUT_SHAPE)])
+        def prog(x):
+            loop_var = np.int32(2)
+            def while_cond(loop_Var, _x):
+                return mb.equal(x=loop_Var, y=np.int32(0))
+
+            def while_body(loop_var, x):
+                # Do reshapes of the input
+                y1 = mb.reshape(x=x, shape=(3, 2, 7))
+                y2 = mb.reshape(x=y1, shape=(7, 2, 3))
+                y3 = mb.reshape(x=y2, shape=(14, 3))
+                y4 = mb.reshape(x=y3, shape=OUTPUT_SHAPE)
+                return mb.add(x=loop_var, y=np.int(-1)), y4
+
+            while_results = mb.while_loop(_cond=while_cond, _body=while_body, loop_vars=(loop_var, x))
+            return while_results[1]
+
+        prev_prog, _, block = apply_pass_and_basic_check(prog, "common::merge_consecutive_reshapes")
```

Comment:
Could you add assert similar to line 1666 and 1667 to validate those reshapes got merged? Thanks!
```
        assert get_op_types_in_program(prev_prog) == [xxx]
        assert get_op_types_in_program(prog) == [xxx]
```

Reply:
I've added those asserts now. Note that I had to add a `recurse` parameter to `get_op_types_in_program` in order for it to return the ops inside the while body block.